src/background/utils.ts

```diff
@@ -37,20 +37,22 @@ export async function getTabIdWithSearchOpen(
 export async function getTabsInCurrentWindow() {
   // should it return the current tab??
   const tabs = await browser.tabs.query({ currentWindow: true });
-  return tabs
-    .filter((tab) => {
-      // will all pages have a title?
-      return tab.id && tab.id !== browser.tabs.TAB_ID_NONE && tab.url;
-    })
-    .map(({ id, favIconUrl, title, url }) => {
+  const results: TabData[] = [];
+
+  for (let i = 0; i < tabs.length; i++) {
```

Comment:
To squeeze out even more performance, you could create a variable instead of `tabs.length` here
```suggestion
const tabNum = tabs.length;
  for (let i = 0; i < tabNum; i++) {
```